OPCB/functions/BADCO_class_check.sqf

```diff
@@ -1,49 +1,37 @@
-BADCO_role_check = 
-{
-	_unit = _this select 0;
-	_role = _this select 1;
-	_vehicle = _this select 2;
-	_unitType = typeOf _unit;
-	_attack = _unit getVariable ["SOAR",0];
-	_attackVehicle = count getPylonMagazines _vehicle;
+params ["_unit", "_role", "_vehicle"];
+_unitType = typeOf _unit;
+_attack = _unit getVariable ["SOAR",0];
+_attackVehicle = count getPylonMagazines _vehicle;
+
+if ((typeof _vehicle ==  "NonSteerable_Parachute_F") || (typeof _vehicle == "Steerable_Parachute_F")) exitWith {}; 
 
-	if ((typeof _vehicle ==  "NonSteerable_Parachute_F") || (typeof _vehicle == "Steerable_Parachute_F")) exitWith {}; 
-	
-	if (_role == "Driver") then 
-	{
-		if (_vehicle isKindOf "Plane") then 
-		{
-			if (_unitType != "rhsusf_airforce_jetpilot") then 
-			{
-				_vehicle engineOn false;
-				moveOut _unit;
-				if (true) exitWith {hint "You must be a jet pilot to fly this"};				
-			};
-			
+if (_role == "Driver") then 
+{
+	if (_vehicle isKindOf "Plane") exitWith {
+		if (_unitType != "rhsusf_airforce_jetpilot") exitWith {
```

Comment:
We should parametrise these classnames

Reply:
I don't see how we could replace the unitTypes with parameters. The caller event is universal for all vehicles. We don't know what type of vehicle they want to enter beforehand.